# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Emeka-techDev/ml-starter/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

Selected Lane: Lane 2
Reason : it helps assist in decision making on what pages needs review first and provide a reasons why the pages should be reviewed.



In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

**Decision the work improves?**

This work assist in prioritizing pages that need review and saving large amount of time and limited resources that would go into impractical review of every page

**Who acts on it?**

Content writer and website reviewers

**Cost of wrong recommendation?**

time and resources are lost reviewing and updating the wrong site, site that actually need update and review might be ommitted, overall visible of the platform does not change


In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [2]:
import pandas as pd

url = "https://huggingface.co/datasets/FlyRank/internship-starter/resolve/main/content_refresh_anonymized.csv"

df = pd.read_csv(url)

df.head()

print(f"Unique Number of pages: {df['content_id'].nunique():,}")

df.shape
print(
    f"Declining pages: {df['is_declining'].sum():,} "
    f"out of {len(df):,}"
)

# declining_rate = df["is_declining"].mean() * 100

# print(f"Declining pages: {declining_rate:.1f}%")
print(f"Total search volume: {df['search_volume'].sum():,.0f}")

Unique Number of pages: 30,000
Declining pages: 5,563 out of 30,000
Total search volume: 4,374,350


The starter file contains 30,000 pages, which makes reviewing each page individually impractical and creates a clear need for prioritization. Of those pages, 5,563 (18%) are declining, this suggest a meaningful subset of the content require review. The dataset contain a total search volume of 4,374,350 which indicates that the content represents a substantial amount of search demand, making the question which page should be reviewed first substantial valuable.
  

In [8]:
#Signal 1 -- Declining
df["declining_score"] = df["is_declining"].astype(int) * 3

# Signal 2 -- Staleness
df["stale_score"] = (df["days_since_last_update"] > 180).astype(int) * 2
median_search_volume = df["search_volume"].median()

#Signal 3 -- Search Demand
df["demand_score"] = (df["search_volume"] > median_search_volume).astype(int) * 2

#Signal 4 -- Ranking Opportunity
position_points = {
    "page_1" : 2,
    "top_3": 2,
    "striking": 1,
    "page_3_5": 1,
    "deep": 0
}

df["position_score"] = (df["position_tier"].map(position_points).fillna(0))

df["baseline_score"] = (df["declining_score"] + df["stale_score"] + df["demand_score"] + df["position_score"])

baseline_queue = df.sort_values(by="baseline_score", ascending=False)

# baseline_queue.head(100)
baseline_queue[
    [
        "content_id",
        "baseline_score",
        "is_declining",
        "days_since_last_update",
        "search_volume",
        "position_tier"
    ]
].head(20)


,content_id,baseline_score,is_declining,days_since_last_update,search_volume,position_tier
6679,content_60e82604d07a,7,True,104,20.0,page_1
22244,content_38ca5024c34b,7,True,104,70.0,page_1
837,content_ebab0744ca5a,7,True,20,260.0,page_1
13286,content_5258796a68ea,7,True,104,210.0,page_1
3103,content_e32ec994d05a,7,True,14,50.0,page_1
27842,content_84e25eb7e6cd,7,True,22,210.0,page_1
27840,content_ced8288da781,7,True,20,30.0,page_1
9392,content_a05de90216c2,7,True,7,90.0,page_1
9814,content_a817704b744d,7,True,104,20.0,page_1
9815,content_b636045f0a0a,7,True,20,40.0,page_1


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.